### DISCUSSION TITLE
Offline Licensing: How to configure dataset?

### DISCUSSION THREAD

**Discussion by artem-denysenko-tw** [keygen-sh/community#80] [Q&A] [answered]
Hi,
I want to change the default embedded dataset to the custom one. I can't find where I could do this. Is there any place on the Keygen UI where I could affect the dataset that will be embedded in the newly generated license key?
I'm using the Ed25519 signing scheme for encryption. I can't figure out, how to include the extra data in the license key payload in case I would like to analyze offline some other features besides the expiration date.

Thanks.

**Comment by ezekg** [comment, marked as answer]
When creating the license key, whatever the license's "key" is set to during creation is what is signed and encoded into the final key.

```diff
curl -X POST 'https://api.keygen.sh/v1/accounts/<account>/licenses' \
  -H 'Content-Type: application/json' \
  -H 'Accept: application/json' \
  -u 'token:<token>' \
  -d '{
        "data": {
          "type": "licenses",
          "attributes": {
+           "key": "{ \"some\": \"data\", \"to\": [\"be\", \"signed\"] }"
          },
          "relationships": {
            "policy": {
              "data": { "type": "policies", "id": "<policy>" }
            }
          }
        }
      }'
```

Would result in, for example:

```diff
{
  "data": {
    "id": "ab5f7b72-ef6e-4fc6-856d-02d38e01c9a5",
    "type": "licenses",
    "attributes": {
      "name": null,
+     "key": "key/eyAic29tZSI6ICJkYXRhIiwgInRvIjogWyJiZSIsICJzaWduZWQiXSB9.nesDnXvk9LCweEP5Dys6qJv5xWkHGIwcNX9PqUZ5cX7uwohKFmwF3Z3rV0ZJPJxxFFCoF3TcV9EfNGBM7G5aDw==",
      "expiry": null,
      ...
    },
    "relationships": {
      ...
    }
  }
}
```

Within the UI, set the license's "key" (which will change to "signing data" if the policy implements a cryptographic scheme).

**Reply by artem-denysenko-tw** [reply]
Thank you! It turned out that everything was so simple, but it was not obvious to me :-)
